internal/mailer/template.go

```diff
@@ -105,16 +106,17 @@ func (m *TemplateMailer) InviteMail(user *models.User, otp, referrerURL string)
 }
 
 // ConfirmationMail sends a signup confirmation mail to a new user
-func (m *TemplateMailer) ConfirmationMail(user *models.User, otp, referrerURL string) error {
+func (m *TemplateMailer) ConfirmationMail(user *models.User, otp, referrerURL string, externalURL *url.URL) error {
 	redirectParam := encodeRedirectParam(referrerURL)
-	fragment := "token=" + user.ConfirmationToken + "&type=signup" + redirectParam
-	url, err := getSiteURL(referrerURL, m.Config.API.ExternalURL, m.Config.Mailer.URLPaths.Confirmation, fragment)
+
+	path, err := getPath(m.Config.Mailer.URLPaths.Confirmation, "token="+user.ConfirmationToken+"&type=signup"+redirectParam)
```

Comment:
Since we're touching this, maybe use `url.Values` to build the query params instead of string concatenation.